network/hybridNetwork.go

```diff
@@ -48,7 +48,19 @@ func NewHybridP2PNetwork(log logging.Logger, cfg config.Local, datadir string, p
 	p2pcfg.NetAddress = cfg.P2PHybridNetAddress
 	p2pcfg.IncomingConnectionsLimit = cfg.P2PHybridIncomingConnectionsLimit
 	identityTracker := NewIdentityTracker()
-	p2pnet, err := NewP2PNetwork(log, p2pcfg, datadir, phonebookAddresses, genesisID, networkID, nodeInfo, &identityOpts{tracker: identityTracker})
+
+	var subnetMeshCreator MeshCreator = meshCreator
+	var hybridMeshCreator MeshCreator = &noopMeshCreator{}
+	_, isHybridMeshCreator := meshCreator.(*HybridRelayMeshCreator)
+	if meshCreator == nil && cfg.IsHybridServer() || isHybridMeshCreator {
```

Comment:
No differentiation in behavior between the two sides of this OR?

Reply:
correct, there are the following cases:
1. old behavior, use base meshing when this node is not a server. Since there are only two exported mesh creators (`BaseMeshCreator` and `HybridRelayMeshCreator`), we can only check for the latter one. Both WS and P2P constructors have (imo) reasonable default to use `BaseMeshCreator` if nothing provided.
2. new behavior if it is a server, or the caller explicitly gave us `HybridRelayMeshCreator`